### PR TITLE
Feature/ Expand a story and copy story url in board V2

### PR DESCRIPTION
**Pull request by kostadriano**
Ref: [#story-22724](https://central.cm42.io/projects/central-board-v2#story-22724)

Expand a Story: 
cm42 central 1

Copy Story URL to clipboard:
cm42 central 2

### CODE REVIEW

File: app/assets/javascripts/actions/story.js [hunk 1]
```diff
@@ -4,3 +4,8 @@ export const receiveStories = (stories) => ({
   type: actionTypes.RECEIVE_STORIES,
   data: stories
 });
+
+export const toggleStory = (id) => ({
+  type: actionTypes.TOGGLE_STORY,
+  id
+})
```
Comment (negreirosleo) [inline app/assets/javascripts/actions/story.js]: missing ; here

File: app/assets/javascripts/components/story/StoryItem.js [hunk 2]
```diff
@@ -1,15 +1,31 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import CollapsedStory from './CollapsedStory';
+import ExpandedStory from './ExpandedStory'
+import { connect } from 'react-redux'
+import { toggleStory } from '../../actions/story'
```
Comment (negreirosleo) [inline app/assets/javascripts/components/story/StoryItem.js]: missing ; on the imports

File: app/assets/javascripts/reducers/columns/done.js [hunk 3]
```diff
@@ -33,7 +34,18 @@ const done = (state = initialState, action) => {
...
+      return {
+        ...state,
+        sprints: (state.sprints.map((sprint) => {
+          return {
+            ...sprint,
+            stories: collapseStory(sprint.stories, action.id)
+          }
+        })
+        )
+      };
```
Comment (negreirosleo) [inline app/assets/javascripts/reducers/columns/done.js]: wouldnt be better to fomat like 
           })
        )};
??

File: app/assets/stylesheets/new_board/_story.scss [hunk 4]
```diff
@@ -1,161 +1,179 @@
...
-      padding: 7px 8px;
       background-color: $story-background-color;
-      border-bottom: 1px solid $story-border-bottom-color;
-      border-top: 1px solid $story-border-top-color;
       min-height: 37px;
       cursor: move;
       align-items: center;
       font-size: 12px;
-
+
```
Comment (negreirosleo) [inline app/assets/stylesheets/new_board/_story.scss]: remove whitespace

File: app/assets/javascripts/reducers/columns/done.js (hunk 3 above)
Comment (kostadriano) [reply app/assets/javascripts/reducers/columns/done.js]: this `)` is to separate the map method in line 41 just for better understanding.

File: app/assets/javascripts/components/story/CollapsedStory/index.js [hunk 5]
```diff
@@ -14,28 +14,34 @@ const classNameStory = (storyType, estimate) => {
...
+          <StoryDescriptionIcon description={story.description} />
+        </div>
+      </StoryPopover>
+
+      <CollapsedStoryInfo story={story} />
+
+      <CollapsedStoryStateActions story={story} />
+    </div>
+  )
+}
```
Comment (negreirosleo) [inline app/assets/javascripts/components/story/CollapsedStory/index.js]: missing ;

File: app/assets/javascripts/components/story/ExpandedStory/index.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ExpandedStoryHistoryLocation from './ExpandedStoryHistoryLocation'
+import ExpandedStoryControls from './ExpandedStoryControls'
```
Comment (negreirosleo) [inline app/assets/javascripts/components/story/ExpandedStory/index.js]: missing ; here too

File: app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryControls.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+const ExpandedStoryControls = (props) => {
+  const { onCancel } = props;
+
+  return (
+    <div className="form-group Story__controls">
+      <input className="cancel" onClick={onCancel} type="button" value="Cancel" />
```
Comment (talyssonoc) [inline app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryControls.js]: We need to use `I18n` here.

File: app/assets/javascripts/reducers/columns/chillyBin.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import actionTypes from 'actions/actionTypes';
 import * as Story from 'models/beta/story';
 import _ from 'underscore';
+import { collapseStory } from '../story';
```
Comment (talyssonoc) [inline app/assets/javascripts/reducers/columns/chillyBin.js]: What do you think about renaming this to `toggleStory` as well?

File: spec/javascripts/components/story/story_item_spec.js [hunk 9]
```diff
@@ -1,13 +1,20 @@
...
+  it('renders the StoryItem component within a Collapsed Story', () => {
+    const story = storyFactory({ collapsed: true });
     const wrapper=shallow(<StoryItem story={story} />);
 
     expect(wrapper.find('CollapsedStory')).toExist();
   });
+
+  it('renders the StoryItem component within a Expanded Story', () => {
+    const story = storyFactory({ collapsed: false });
+    const wrapper=shallow(<StoryItem story={story} />);
```
Comment (talyssonoc) [inline spec/javascripts/components/story/story_item_spec.js]: Can we add some spacing here?

File: spec/javascripts/reducers/column_backlog_spec.js [hunk 10]
```diff
@@ -63,14 +81,49 @@ describe('Backlog Column reducer', () => {
         startDate: "2018-09-03T16:00:00",
         iterationLength: 1,
         defaultVelocity: 2,
-      },
+      }
     }
-  }
+  };
+
+  function toggleStoryAction(id) {
```
Comment (talyssonoc) [inline spec/javascripts/reducers/column_backlog_spec.js]: You can import it from here https://github.com/Codeminer42/cm42-central/pull/417/files#diff-a70c128fa112ed9628c46ebccc4a8fd2R22

File: spec/javascripts/reducers/column_backlog_spec.js [hunk 11]
```diff
@@ -63,14 +81,49 @@ describe('Backlog Column reducer', () => {
...
+    return {
+      type: actionTypes.TOGGLE_STORY,
+      id
+    }
+  };
+
+  describe("Toggle a story", () => {
+    it("expand", () => {
```
Comment (talyssonoc) [inline spec/javascripts/reducers/column_backlog_spec.js]: We could add some contexts here, like `when story is collapsed` and `when story is expanded` around the `it` to better readability 😄

**Pull request merged**